lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -56,6 +56,68 @@ removeNode = (node) ->
   removedNode = node.parentNode.removeChild(node)
   triggerEvent('page:after-node-removed', removedNode)
 
+updateHead = (doc) ->
+  loadedAssets = extractTrackAssets document
+  fetchedAssets = extractTrackAssets doc
+
+  updateLinkTags(loadedAssets, fetchedAssets)
+  updateScriptTags(loadedAssets, fetchedAssets)
+
+extractTrackAssets = (doc) ->
+  for node in doc.querySelector('head').childNodes when node.getAttribute?('data-turbolinks-track')?
+    node
+
+filterForNodeType = (nodeList, nodeType) ->
+  (node for node in nodeList when node.nodeName == nodeType)
+
+insertScript = (scriptNode) ->
```

Comment:
Mhm! Took me a bit of time to figure our what was going on with this and arrive at this solution, but basically:
- `document.appendChild(otherDocument.querySelector('script#some-id'))` **does not cause the script to be evaluated**, for some reason.
- using `document.cloneNode` also seems to have this problem, for some reason.
- generating a new script tag with identical attributes and appending it works fine.